testing/supporting_docs/ui_selectors_and_their_importance.md

```diff
@@ -0,0 +1,178 @@
+    android:layout_alignParentBottom="true"
+    android:layout_alignParentStart="true" />
+```
+
+
+![alt text](./images/authenticate_button_in_appium_inspector.png "Accessibility Id Now Available to Use")
+
+### Xamarin (AutomationId Example)
+Xamarin elements have a unique attribute called AutomationId, which can be used to set UI selectors 
+across both iOS and Android platforms.  This can be set in either the XAML code or in the C# backend 
+code.  
+
+```cs
+// C# Code
+var button = new Button{
+    Text = "Authenticate"
+    AutomationId = "authenticate_button"
+}
+```
+
+
+```xml
+<ic:CircleImage AutomationId="my_image" 
+    WidthRequest="100" 
+    HeightRequest="100"
+    Aspect="AspectFill" />
+```
+
+
+
+![img text](./images/accessibilityid_set_from_automationid.png "Resulting in AccessibilityId and Name Attributes Being set as Shown in Appium Inspector")
+
+### Cordova (id Example)
+
+Unlike Android, iOS and Cordova applications with can be tested using Appiums native context, webview
+context should be used to test cordova/ionic applications.  When testing applications using the webview
+context, web based targeting selectors should be used.  For example in the Java-client the full Selenium
+WebDriver API can be used, while in the Nodejs Webdriverio library the standard webdriverio selector 
+targeting methods can be used.  Note that in both of these libraries, the standard web based selector or
+locator strategies are not available when using the native context.
+
+In order to prepare Appium to test a cordova/ionic application in the desied capabilities section,
+`autoWebview` shoul be included and set to `true`.  This will immediatley prepare the test to run in a
+webview context.  Note, the native view selector or locator stategies will now be unavailable.
+
+The following is an example of using standard id attributes in an ionic based application.  With these
+ids specified, when running in webview context, Appium can use the standard web app based selectors for
+targeting these elements.  
+
+```html
+<ion-card>
+
+    <ion-card-content style="text-align: center">
+        <ion-item>
+            <ion-avatar item-start>
+            <img src="./assets/imgs/ag-logo.png">
+            </ion-avatar>
+            <h3 id='fullName'>{{profile.firstName}} {{profile.lastName}}</h3>
+            <p>Full Name</p>
+        </ion-item>
+        <ion-item>
+            <h3 id='email'> {{profile.email}} </h3>
+            <p> Email </p>
+        </ion-item>
+        <ion-item>
+            <h3 id='username'> {{profile.username}} </h3>
+            <p> Username </p>
+        </ion-item>
+    </ion-card-content>
+</ion-card>
+```
+
+One dependency when running uder webview is the compatibility of the version of chromedriver that is
+being used with Appium.  A version is shipped with the Appium, however this is not always compatible
+with the chrome browser running in the android device.  In this case an error can be thrown when 
```

Comment:
@twnolan consistency should be persisted with the word 'android' across the whole document, e.g. starting with either lower or upper case.